From 643167423858052c26b2dfaf332c1ec0b472ea7a Mon Sep 17 00:00:00 2001
From: Yannick Brosseau <yannick.brosseau@gmail.com>
Date: Fri, 2 Dec 2011 21:13:32 -0500
Subject: staging: lttng: Fix recent modifications to string_from_user
 operation

Include: a fix for a recently introduced change: obviously max_t should
be used instead of min_t here. Also, a likely should apply to the result
of the comparison, not the variable per se.

Signed-off-by: Yannick Brosseau <yannick.brosseau@gmail.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/lttng/probes/lttng-events.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/lttng/probes/lttng-events.h b/drivers/staging/lttng/probes/lttng-events.h
index ff6273f..d486994 100644
--- a/drivers/staging/lttng/probes/lttng-events.h
+++ b/drivers/staging/lttng/probes/lttng-events.h
@@ -347,7 +347,7 @@ static __used struct lttng_probe_desc TP_ID(__probe_desc___, TRACE_SYSTEM) = {
 #undef __string_from_user
 #define __string_from_user(_item, _src)					       \
 	__event_len += __dynamic_len[__dynamic_len_idx++] =		       \
-		min_t(size_t, strlen_user(_src), 1);
+		max_t(size_t, strlen_user(_src), 1);
 
 #undef TP_PROTO
 #define TP_PROTO(args...) args
@@ -557,7 +557,7 @@ __assign_##dest##_2:							\
 			(void) __typemap.dest;				\
 		lib_ring_buffer_align_ctx(&__ctx, ltt_alignof(__typemap.dest));\
 		__ustrlen = __get_dynamic_array_len(dest);		\
-		if (likely(__ustrlen) > 1) {				\
+		if (likely(__ustrlen > 1)) {				\
 			__chan->ops->event_write_from_user(&__ctx, src,	\
 				__ustrlen - 1);				\
 		}							\
-- 
1.7.9

